functorch/experimental/_map.py

```diff
@@ -320,66 +356,25 @@ def map_fake_tensor_mode(mode, f, num_mapped, *args):
         return map_dense(f, num_mapped, *args)
 
 
-@map_impl.py_impl(DispatchKey.Functionalize)
-def map_func(f, num_mapped, *args):
-    reapply_views = torch._C._functionalization_reapply_views_tls()
+@map_impl.py_functionalize_impl()
+def map_functionalize(ctx, f, num_mapped, *args):
     xs = args[:num_mapped]
     pos_args = args[num_mapped:]
-    unwrapped_xs = _unwrap_all_tensors_from_functional(xs, reapply_views=reapply_views)
-    unwrapped_args = _unwrap_all_tensors_from_functional(
-        pos_args, reapply_views=reapply_views
-    )
-    mode = "mutations_and_views" if reapply_views else "mutations"
+    unwrapped_xs = ctx.unwrap(xs)
+    unwrapped_args = ctx.unwrap(pos_args)
+    wrapped_fn = ctx.wrap_function(f)
 
-    with _ExcludeDispatchKeyGuard(DispatchKeySet(DispatchKey.Functionalize)):
-        functional_map_fn = functionalize(f, remove=mode)
+    with ctx.redispatch():
```

Comment:
nit: suggesting some more descriptive names:
- unwrap -> ctx.unwrap_tensors
- wrap -> ctx.wrap_tensors
- wrap_function -> ctx.functionalize (the functionalization API)
- redispatch() -> redispatch_to_next()

Reply:
I like all these better, thanks - will change!